### ISSUE TITLE
cucumber trapping SIGINT behavior

### ISSUE THREAD

**Issue by akostadinov** [cucumber/cucumber#27]
Hello, I observe the following behavior.
= regular cucumber vs SIGINT
* test is running
* hit ctrl+c
* test continue to be running and program exits only after it completes fully
* hit ctrl+c again
* cucumber exits without any clean-up taking place

= cucumber with `Signal.trap('SIGINT') { exit(false) }`
* test running
* hit ctrl+c
* test terminates and cucumber switches to After hook

I think that the latter behavior makes much more sense. If user wants to interrupt, then ideally that will happen as soon as possible while After hook is executed to clean-up any underlying test environment. This is current behavior for SIGTERM.

So I was wondering what was the idea behind default SIGINT trap behavior. Why not keep same as SIGTERM? Not a big issue for me as I have a workaround. But would be nice to make behavior more convenient OOB for other users.

btw very nice cucumber takes care to run `After` hook on SIGTERM. I made some effort to put clean-up in `at_exit` when `After` is skipped but it seems `After` is never skipped. Or was the idea to have single Ctrl+C terminate after scenario finish, second Ctrl+C to drop into `After` hook? In such case it is not working like that.

**Comment by akostadinov** [comment]
closing in favor of https://github.com/cucumber/cucumber-ruby-core/issues/108
